agent/consul/config_endpoint.go

```diff
@@ -449,23 +450,23 @@ func (c *ConfigEntry) ResolveServiceConfig(args *structs.ServiceConfigRequest, r
 				}
 			}
 
-			// Then store upstreams inferred from service-defaults
-			if serviceConf != nil && serviceConf.Connect != nil {
-				for sid := range serviceConf.Connect.UpstreamConfigs {
-					seenUpstreams[structs.ServiceIDFromString(sid)] = struct{}{}
-				}
-			}
-
-			// usConfigs stores the opaque config map for each upstream and is keyed on the upstream's ID.
-			usConfigs := make(map[structs.ServiceID]map[string]interface{})
-
+			// Then store upstreams inferred from service-defaults and mapify the overrides.
 			var (
+				upstreamConfigs  = make(map[structs.ServiceID]*structs.UpstreamConfig)
 				upstreamDefaults *structs.UpstreamConfig
-				upstreamConfigs  map[string]*structs.UpstreamConfig
+				// usConfigs stores the opaque config map for each upstream and is keyed on the upstream's ID.
+				usConfigs = make(map[structs.ServiceID]map[string]interface{})
 			)
-			if serviceConf != nil && serviceConf.Connect != nil {
-				if serviceConf.Connect.UpstreamDefaults != nil {
-					upstreamDefaults = serviceConf.Connect.UpstreamDefaults
+			if serviceConf != nil && serviceConf.UpstreamConfig != nil {
+				for _, override := range serviceConf.UpstreamConfig.Overrides {
+					if override.Name == "" {
+						continue // skip this impossible condition
```

Comment:
Why is it impossible? Feels like it either should be handled (at least logged) or not tested for....

Reply:
Earlier validation before the config entry was accepted should require the name field to be set.

Reply:
logged